kivy/animation.py

```diff
@@ -455,11 +421,64 @@ def cancel_property(self, widget, prop):
                 not self.anim2.have_properties_to_animate(widget)):
             self.cancel(widget)
 
-    def on_anim1_start(self, instance, widget):
+    def have_properties_to_animate(self, widget):
+        return (self.anim1.have_properties_to_animate(widget) or
+                self.anim2.have_properties_to_animate(widget))
+
+    @property
+    def animated_properties(self):
+        # TODO: after kivy drops python2 support, this code can be optimized.
+        #       for example: using ChainMap
+        d = dict(self.anim1.animated_properties)
+        d.update(self.anim2.animated_properties)
+        return d
```

Comment:
Might be better to not use `ChainMap`. Because if we re-implement it like this:

```python
    @property
    def animated_properties(self):
        return ChainMap(self.anim2.animated_properties, self.anim1.animated_properties)
```

the user will be able to write `anim2.animated_properties`.

Reply:
in both case it's an object that is returned to the user, that they can write to, but that doesn't mean they can write to animation.animated_properties, unless the property defines a setter.

Reply:
Oh sorry, I meant this part

> Lookups search the underlying mappings successively until a key is found. In contrast, writes, updates, and deletions only operate on the first mapping.

in the [doc](https://docs.python.org/3/library/collections.html#collections.ChainMap).

To prevent the user from writing to underlying dictionaries(`anim1.animated_properties` and `anim2.animated_properties`), we need to impelement it like either of

```python
@property
def animated_properties(self):
    return {**self.anim1.animated_properties, **self.anim2.animated_properties, }
```

or 

```python
@property
def animated_properties(self):
    return ChainMap({}, self.anim2.animated_properties, self.anim1.animated_properties)
```

Reply:
ok, that makes sense indeed. I guess the second one is a bit more efficient, as it doesn't unnecessarily copy data around, while still allowing to update the values for this animation (as they'll be saved in the first dict, they'll override values for the same key in the other dicts), and also they'll reflect updates to the animated properties of the underlying animations, in case they happen (and unless an overriding value has been put in the first dict, of course).